submitit/core/test_async.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+
+import asyncio
+from pathlib import Path
+
+import pytest
+
+from . import submission, utils
+from .test_core import FakeExecutor, _three_time
+
+
+@pytest.mark.asyncio
+async def test_result(tmp_path: Path):
+    executor = FakeExecutor(folder=tmp_path)
+    job = executor.submit(_three_time, 8)
+    result_task = asyncio.get_event_loop().create_task(job.async_result())
```

Comment:
I'm also puzzled by this, it looks way more complicated than your readme example.

Do you need explicitly the event loop ? In that case, I'd prefer you to use the `event_loop` fixture.
Otherwise it seems that `@pytest.mark.asyncio` should be enough to call await on the `job.async_result()`.

Reply:
so
`await job.async_results()`
will wait inline and block until the job results are available. In the test, there is no real async job running in the back, so it will never happen unless we call: `submission.process_job(folder=job.paths.folder)`.

`loop.create_task` will send the coroutine to the background, so we can then execute the submission and ln21 get the result that is available.

This is purely like this because we are in a test and the job is not actually running in the background. I'll add comments. We would never do this in real code as the job always represent an async process.